accounts-db/src/accounts_db.rs

```diff
@@ -6755,140 +6705,170 @@ impl AccountsDb {
+                            last_update = now;
+                        }
+                        thread::sleep(Duration::from_millis(500))
+                    }
+                })
+                .expect("spawn thread");
+            for thread_handle in thread_handles {
+                let Ok(thread_accum) = thread_handle.join() else {
+                    exit_logger.store(true, Ordering::Relaxed);
+                    panic!("index generation failed");
+                };
+                total_accum.insert_us += thread_accum.insert_us;
+                total_accum.num_accounts += thread_accum.num_accounts;
+                total_accum.accounts_data_len += thread_accum.accounts_data_len;
+                total_accum
+                    .zero_lamport_pubkeys
+                    .extend(thread_accum.zero_lamport_pubkeys);
+                total_accum.all_accounts_are_zero_lamports_slots +=
+                    thread_accum.all_accounts_are_zero_lamports_slots;
+                total_accum
+                    .all_zeros_slots
+                    .extend(thread_accum.all_zeros_slots);
+                total_accum.num_did_not_exist += thread_accum.num_did_not_exist;
+                total_accum.num_existed_in_mem += thread_accum.num_existed_in_mem;
+                total_accum.num_existed_on_disk += thread_accum.num_existed_on_disk;
+                total_accum.lt_hash.mix_in(&thread_accum.lt_hash);
+            }
+            // Make sure to join the logger thread *after* the main threads.
+            // This way, if a main thread errors, we won't spin indefinitely
+            // waiting for the logger thread to finish (it never will).
+            logger_thread_handle.join().expect("join thread");
+        });
         index_time.stop();
 
+        {
+            // Update the index stats now.
+            let index_stats = self.accounts_index.bucket_map_holder_stats();
+
+            // stats for inserted entries that previously did *not* exist
+            index_stats.inc_insert_count(total_accum.num_did_not_exist);
+            index_stats.add_mem_count(total_accum.num_did_not_exist as usize);
+
+            // stats for inserted entries that previous did exist *in-mem*
+            index_stats
+                .entries_from_mem
+                .fetch_add(total_accum.num_existed_in_mem, Ordering::Relaxed);
+            index_stats
+                .updates_in_mem
+                .fetch_add(total_accum.num_existed_in_mem, Ordering::Relaxed);
+
+            // stats for inserted entries that previously did exist *on-disk*
+            index_stats.add_mem_count(total_accum.num_existed_on_disk as usize);
+            index_stats
+                .entries_missing
+                .fetch_add(total_accum.num_existed_on_disk, Ordering::Relaxed);
+            index_stats
+                .updates_in_mem
+                .fetch_add(total_accum.num_existed_on_disk, Ordering::Relaxed);
+        }
```

Comment:
this could also land as a IndexGenerationAccumulator's function like
`fn add_to_stats(&self, stats: &mut Stats)`